### PR TITLE
[EventHub] `az eventhubs namespace application-group policy remove`: Add upcoming breaking change notification

### PR DESCRIPTION
**Pull request by schaudhari6254888**
**Related command**
`az eventhubs namespace application-group policy remove`

**Description**
Removing parameters metric-id, rate-limit-threshold from --throttling-policy-config object only for `az eventhubs namespace application-group policy remove` command.
Additionally changing `throttling-policy-config` name to `policy`,  for `az eventhubs namespace application-group policy remove` command. 
The change will release in Upcoming breaking change cycle i.e May.

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
EventHub

File: src/azure-cli/azure/cli/command_modules/eventhubs/custom.py [hunk 1]
```diff
@@ -622,6 +622,9 @@ def cli_remove_appgroup_policy(client, resource_group_name, namespace_name, appl
 
     appGroup = client.get(resource_group_name, namespace_name, application_group_name)
 
+    logger.warning(
+        'Parameter metric-id,rate-limit-threshold would not be required for a command eventhub application-group policy remove in future release '
+        'and the new policy remove cmdlets will br contain only name.')
```
Comment (damodaravadhani) [inline src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:627]: "the policy remove policy cmdlets will do removals based on policy name"

**Comment by schaudhari6254888** [comment]
Can someone please review & merge this pull request. Only contain the breaking change alert message

File: src/azure-cli/azure/cli/command_modules/eventhubs/custom.py [hunk 2]
```diff
@@ -621,7 +621,8 @@ def cli_remove_appgroup_policy(client, resource_group_name, namespace_name, appl
     from azure.cli.core import CLIError
 
     appGroup = client.get(resource_group_name, namespace_name, application_group_name)
-
+    logger.warning(
+        'the policy remove policy cmdlets will do removals based on policy name')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:625]: The sentence sounds awkward to me...
```suggestion
        'This operation will do removals based on policy name')
```

**Comment by evelyn-ys** [comment]
If the "breaking change" you mentioned is "metric-id,rate-limit-threshold would not be required", then I don't think it breaks anything. So you can just modify the title & description with Add warning for balabala

**Comment by evelyn-ys** [comment]
When will you deprecate these parameters? The code change seems nothing to do with the breaking change...

Just FYI, CLI only allows breaking change twice a year and next breaking change window is May release

**Comment by schaudhari6254888** [comment]
> --throttling-policy-config

Inside the --throttling-policy-config object these parameters are present for removal policy.
In future these 2 parameters are not needed for deletion.
I want to do this change in this "breaking change cycle" only.

**Comment by evelyn-ys** [comment]
So this PR is not related with this breaking change, right? Can you modify the PR title and description?

**Comment by schaudhari6254888** [comment]
> So this PR is not related with this breaking change, right? Can you modify the PR title and description?

This PR, I am only giving Message notification.

File: src/azure-cli/azure/cli/command_modules/eventhubs/custom.py [hunk 3]
```diff
@@ -621,7 +621,8 @@ def cli_remove_appgroup_policy(client, resource_group_name, namespace_name, appl
     from azure.cli.core import CLIError
 
     appGroup = client.get(resource_group_name, namespace_name, application_group_name)
-
+    logger.warning(
+        'This operation will do removals based on policy name')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:625]: I don't think with this message customers are able to realize there will be breaking change in the future......

It just says policy name will be used for removals

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/eventhubs/custom.py:625]: If you want to notify that you won't accept `metric-id,rate-limit-threshold` in the future, pls say it explicitly

Comment (schaudhari6254888) [reply src/azure-cli/azure/cli/command_modules/eventhubs/custom.py]: Hey @evelyn-ys , Sorry for the inconvenience.
Does the Description & warning message makes the good sense now?
Any suggestions are always welcome.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**